profiles/views.py

```diff
@@ -52,40 +52,32 @@ class SavedCompaniesCreate(CreateAPIView):
     serializer_class = SavedCompanySerializer
     pagination_class = ForumPagination
 
-    def post(self, request):
+    def post(self, request, *args, **kwargs):
         user = request.user
-        pk = request.data.get("company_pk")
-
-        # Check if the company is already in the user's saved list
-        if SavedCompany.objects.filter(user=user, company_id=pk).exists():
-            saved_company_destroyer = SavedCompaniesDestroy()
-            return saved_company_destroyer.destroy(request, pk)
-
-        serializer = SavedCompanySerializer(
-            data={"company": pk, "user": user.id}
-        )
-        if serializer.is_valid():
-            serializer.save()
-            return Response({"Company added": serializer.data})
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        pk = request.data.get("company")
+
+        saved_company = SavedCompany.objects.filter(
+            user=user, company_id=pk
+        ).first()
+        if saved_company:
+            return redirect(
+                "profiles:saved_companies_destroy", pk=saved_company.pk
+            )
+        request.data.update({"user": user.id})
+        return self.create(request, *args, **kwargs)
```

Comment:
Nope.

1. Don't read `request.data` directly (we have serializers for this).
2. Don't manipulate (change) `request.data`. **EVER!**
3. Don't do redirects in REST API.